letta/functions/schema_generator.py

```diff
@@ -1,8 +1,10 @@
 import inspect
 import typing
-from typing import Any, Dict, Optional, Type, get_args, get_origin
+from typing import Any, Dict, Optional, Type, Union, get_args, get_origin
 
+from crewai_tools.tools.base_tool import BaseModel as CrewAiBaseModel
```

Comment:
this might error if the extra is not installed?

Reply:
Moved to within the relevant function